### PR TITLE
Auto align rewrite

### PR DESCRIPTION
**Pull request by Leaf-life**
fixed the state michine to use idiling and simplified the logic for calculating the degrees from alignment to include less if statments

### CODE REVIEW

**Comment by Team3200** [comment]
I'm sorry
-Caiden

File: components/autoAlign.py [hunk 1]
```diff
@@ -50,50 +54,69 @@ def toggleShootAfterComplete(self):
...
+                    break
 
-                # If the horizontal offset is within the given tolerance,
-                # finish.
-                else:
-                    log.info("Autoalign complete")
-                    self.driveTrain.setTank(0, 0)
-                    if self.shootAfterComplete:
-                        self.autoShoot.startAutoShoot()
+            log.info("Autoalign complete")
```
Comment (ManiacMoustache) [inline components/autoAlign.py:101]: Line 101 will run every loop, which won't be accurate.

File: components/autoAlign.py [hunk 2]
```diff
@@ -50,50 +54,69 @@ def toggleShootAfterComplete(self):
...
+                    or len(dists) == 2
+                    and dists[0] < self.AbsoluteX
+                    and self.AbsoluteX < dists[1]):
+                    if speed == "PID":
+                        self.speed = self.calc_PID(self.DeviationX)
                     else:
-                        self.speed = self.DumbSpeed
-                    self.next_state_now("adjust_self_left")
+                        self.speed = speed
+                    self.next_state("adjust_self")
```
Comment (ManiacMoustache) [inline components/autoAlign.py:98]: This causes an infinite loop (which we stop in robot.py, but it's better if the component exits by itself)

**Review by ManiacMoustache — CHANGES_REQUESTED** [review]
Better than my original version

**Review by ManiacMoustache — DISMISSED** [review]
+1
Better than mine

**Review by hell0w0rld2 — DISMISSED** [review]
Approved

**Review by ManiacMoustache — APPROVED** [review]
+1 Recent commits have not changed my opinion

**Review by hell0w0rld2 — APPROVED** [review]

**Pull request merged**